src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -361,6 +361,33 @@ using StrBeforeExpression =
     StringExpressionImpl<2, LiftStringFunction<decltype(strBefore)>,
                          StringValueGetter>;
 
+[[maybe_unused]] auto mergeFlagsIntoRegex =
+    [](std::optional<std::string> regex,
+       const std::optional<std::string>& flags) -> IdOrLiteralOrIri {
+  if (!flags.has_value() || !regex.has_value()) {
+    return Id::makeUndefined();
+  }
+  auto firstInvalidFlag = flags.value().find_first_not_of("imsu");
+  if (firstInvalidFlag != std::string::npos) {
+    throw std::runtime_error{absl::StrCat(
+        "Invalid regex flag '",
+        std::string_view{&flags.value()[firstInvalidFlag], 1}, "' found in \"",
+        flags.value(),
+        "\". The only supported flags are 'i', 'm', 's', 'u', and any "
+        "combination of them")};
+  }
+
+  // In Google RE2 the flags are directly part of the regex.
+  std::string result =
+      flags.value().empty()
```

Comment:
Is this function code duplication to something we have inside the `RegexExpression` class? if yes, deduplicate it (some general helper class that can be easily tested called regexHelpers etc.

Reply:
Also when replacing the throws statement with return undef, then they no longer have much in common.